### DISCUSSION TITLE
xr.open_dataset(url) gives NetCDF4 (lru_cache.py) error "oc_open: Could not read url"

### DISCUSSION THREAD

**Discussion by ghost** [pydata/xarray#6742] [Q&A]
This code I use was working about a year ago but today gives me an error:

```
import xarray as xr
url = 'http://www.esrl.noaa.gov/psd/thredds/dodsC/Datasets/NARR/monolevel/uwnd.10m.2000.nc'
ds = xr. open_dataset(url)
```

The Traceback includes the following:

```
File "C:\Users\Codiga_D\AppData\Local\Continuum\miniconda3\envs\EQ\lib\site-packages\xarray\backends\lru_cache.py", line 53, in __getitem__
    value = self._cache[key]
```

and

```
OSError: [Errno -68] NetCDF: I/O failure: b'http://www.esrl.noaa.gov/psd/thredds/dodsC/Datasets/NARR/monolevel/uwnd.10m.2000.nc'

Note:Caching=1
Error:curl error: SSL connect error
curl error details: 
Warning:oc_open: Could not read url
```

I am on Windows using a conda virtual env. I am pretty sure the esrl server hasn't changed from a year ago (it isn't password protected as far as I know) but I suppose that is possible.

Any suggestions?

**Comment by MartinPontius** [comment]
I ran into the same problem. As a temporary fix you could pin the version of the `netCDF4` Python library to 1.5.8 (latest version is 1.6.0). At least this solved it in my case as the problem only occurred with version 1.6.0. 
However, the true cause of the problem is still unclear to me.

**Comment by ghost** [comment]
Thank you MartinPontius for the suggestion. Very helpful. I guess there aren't very many folks who have faced this issue. I was expecting more responses than I have so far had-- only yours.

Unfortunately, I just pinned my netcdf4 to be 1.5.8 as you suggested, and the same error is occurring.

I guess I should create an 'issue' in addition to this 'discussion'... perhaps that will be seen by more folks who could give suggestions.

**Comment by ghost** [comment]
I also should mention that I confirmed the ESRL server is not requiring a password. 

Also, Cathy Smith at PSL (formerly ESRL, I believe) informed me that it has a new name, so it is better to use a URL with the new name in it:

https://psl.noaa.gov/thredds/dodsC/Datasets/NARR/monolevel/uwnd.10m.2000.nc. 

However I tried that and got the same error. 
Which I believe means that any name-related redirections are working fine, and the problem is not related to them.

**Comment by su123454321** [comment]
Hello, today when i try to run my executable to download GFS data,i meet the same problem:Error:curl error: SSL connect error
curl error details: 
Warning:oc_open: Could not read url
Failed to download .nc: [Errno -68] NetCDF: I/O failure: b'https://nomads.ncep.noaa.gov/dods/gfs_0p25/gfs20240926/gfs_0p25_00z'.  i can run the python  script without error, but when i use pyinstaller to package it and run the executable, it gets wrong, what should i do?
